### DISCUSSION TITLE
pnpm deploy fails on workspace project postinstall script...

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [pnpm/pnpm#5329] [Q&A] [answered]
In the following screenshot you see an error when running:

```
pnpm --filter zebra deploy apps/zebra/build-deploy
```

`zebra` is an app and it depends on the workspace project `@libs/mammoth-prisma`.

`@libs/mammoth-prisma` has a postinstall script that is currently failing when triggered indirectly by the pnpm deploy command running mentioned above.

It seems that `@libs/mammoth-prisma` has no dependencies (dev deps?) available to it.

Am I doing something wrong?

![CleanShot 2022-09-09 at 12 50 07@2x](https://user-images.githubusercontent.com/284476/189401104-bd73e915-db99-4a72-b80e-d8c2aa937ac2.png)

**Comment by zkochan** [comment, marked as answer]
When you deploy, `@libs/mammoth-prisma` is installed as a dependency. devDependencies of dependencies are not installed. So if some dependency is used by the postinstall of `@libs/mammoth-prisma`, it should be a prod dependency.

**Reply by jasonkuhrt** [reply]
Thanks @zkochan, IIUC this caveat is particular to `deploy` and not the case for eg. `pnpm install`. We should probably document this. Open to a PR for that?

**Reply by zkochan** [reply]
It is the case for `pnpm install`. If you install `@libs/mammoth-prisma` in a different project, postinstall will fail if it uses a dev dependency.

**Reply by jasonkuhrt** [reply]
Ah! Learning new things _every day_ :)
